ip6t_REJECT patch to send icmpv6 unreachable type 1 (destination unrechable)  
code 5 (source address failed ingress/egress policy)
ref: http://www.iana.org/assignments/icmpv6-parameters/icmpv6-parameters.xhtml

diff --git a/include/uapi/linux/netfilter_ipv6/ip6t_REJECT.h b/include/uapi/linux/netfilter_ipv6/ip6t_REJECT.h
--- a/include/uapi/linux/netfilter_ipv6/ip6t_REJECT.h
+++ b/include/uapi/linux/netfilter_ipv6/ip6t_REJECT.h
@@ -10,7 +10,8 @@ enum ip6t_reject_with {
 	IP6T_ICMP6_ADDR_UNREACH,
 	IP6T_ICMP6_PORT_UNREACH,
 	IP6T_ICMP6_ECHOREPLY,
-	IP6T_TCP_RESET
+	IP6T_TCP_RESET,
+	IP6T_ICMP6_POLICY_FAIL
 };
 
 struct ip6t_reject_info {
diff --git a/net/ipv6/netfilter/ip6t_REJECT.c b/net/ipv6/netfilter/ip6t_REJECT.c
--- a/net/ipv6/netfilter/ip6t_REJECT.c
+++ b/net/ipv6/netfilter/ip6t_REJECT.c
@@ -211,6 +211,10 @@ reject_tg6(struct sk_buff *skb, const st
 	case IP6T_TCP_RESET:
 		send_reset(net, skb);
 		break;
+	case IP6T_ICMP6_POLICY_FAIL:
+		send_unreach(net, skb, ICMPV6_POLICY_FAIL, par->hooknum);
+		break;
+
 	default:
 		net_info_ratelimited("case %u not handled yet\n", reject->with);
 		break;
